### ISSUE TITLE
Bug: `no-unused-vars` thinks a lowercase name is used in JSX element tag

### ISSUE THREAD

**Issue by alanorozco** [eslint/eslint#15040]
### Environment

Node version: v14.17.5
npm version: v6.14.14
Local ESLint version: Not found
Global ESLint version: Not found
Operating System: darwin 20.6.0

### What parser are you using?

@babel/eslint-parser

### What did you do?

```jsx
const unused = () => {};
function Example() {
  return <unused />;
}
```

### What did you expect to happen?

[According to the JSX spec](https://reactjs.org/docs/jsx-in-depth.html#specifying-the-react-element-type), the output `createElement` call disambiguates between component references (JS) and string tag names (DOM) depending on the first letter of a name being capitalized:

```jsx
// input
<Capitalized />
<lowercase />
```

```js
// output
createElement(Capitalized);
createElement("lowercase");
```

Thus, a lowercase identifier should **not** be considered used when it's only part of a JSX tag.

The previous example ("what did you do?") **should** report a `no-unused-vars` error.

### What actually happened?

The example does not report a `no-unused-vars` error.

### Participation

- [X] I am willing to submit a pull request for this issue.

**Comment by alanorozco** [comment]
This is likely a regression or untested case of #1911

**Comment by mdjermanovic** [comment]
Hi @alanorozco, thanks for the issue!

Can you please share your ESLint configuration?

Also, if you have [`react/jsx-uses-vars`](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/jsx-uses-vars.md) enabled, can you try to disable that rule and see if a `no-unused-var` error will appear after that?

**Comment by alanorozco** [comment]
@mdjermanovic Sure.

### .eslintrc.js

```js
module.exports = {
  root: true,
  parser: "@babel/eslint-parser",
  plugins: ["react"],
  env: {
    es6: true,
    browser: true,
  },
  parserOptions: {
    requireConfigFile: false,
    ecmaVersion: 6,
    jsx: true,
    sourceType: "module",
    babelOptions: {
      presets: ["@babel/preset-react"],
    },
  },
  rules: {
    "react/jsx-uses-vars": 2,
    "no-unused-vars": 2,
  },
};
```
### case
```js
const Used = () => {};
const Unused = () => {};
const lowercase = () => {};

export function Example() {
  return (
    <>
      <Used />
      <lowercase />
    </>
  );
}
```
### output

```
  2:7   error  'Unused' is assigned a value but never used  no-unused-vars
```

### output with `jsx-uses-vars` disabled

```
  1:7   error  'Used' is assigned a value but never used       no-unused-vars
  2:7   error  'Unused' is assigned a value but never used     no-unused-vars
  3:7   error  'lowercase' is assigned a value but never used  no-unused-vars
```

**Comment by mdjermanovic** [comment]
Thanks!

As you've already noticed (and prepared a PR in eslint-plugin-react), this is an issue with the `react/jsx-uses-vars` rule.

`no-unused-vars` itself does not treat JSX identifiers as references because that's React-specific JSX semantics, so this would be, by default, a reported error on the `lowercase` variable, which is observable when you disable `react/jsx-uses-vars` rule.

However, `no-unused-vars` has an additional feature - other rules can mark variables as used, and `no-unused-vars` will not report those variables. In this case, `react/jsx-uses-vars` marks the `lowercase` variable as used.

I'm closing this issue since this isn't a problem in the core rule.

**Comment by alanorozco** [comment]
I encountered this on a larger repo and was initially unaware that this was triggered by the `react` plugin.

Thank you for your help in finding the culprit!
